share/rocm/cmake/ROCMInstallTargets.cmake

```diff
@@ -56,13 +123,50 @@ function(rocm_install_targets)
             PATTERN "*.inl")
     endforeach()
 
-    install(
-        TARGETS ${PARSE_TARGETS}
-        EXPORT ${EXPORT_FILE}
-        RUNTIME DESTINATION ${BIN_INSTALL_DIR}
-        LIBRARY DESTINATION ${LIB_INSTALL_DIR}
-        ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
-
+    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12.0")
```

Comment:
Is this necessary? Can the implementation for 3.10 work on newer cmake? Or is there a functional difference?

Reply:
Devops is on cmake 3.18. Do we need to support older cmake versions?

Reply:
The 3.10 implementation apparently does work for newer CMake, previously I had encountered errors complaining about having a target multiple times in one export file when running the test suite on newer CMake versions. Reordering the installation of NAMELINK_SKIP and NAMELINK_ONLY apparently fixed this issue.

Reply:
We are on cmake 3.16.8 as of now:

docker run -it --rm compute-artifactory.amd.com:5000/rocm-plus-docker/ubuntu-18.04-bld:latest bash
root@7b027ce1e7bb:/src# cmake --version
cmake version 3.16.8

CMake suite maintained and supported by Kitware (kitware.com/cmake).

Reply:
First, to pass CI it needs to work with cmake 3.10 or 3.20(the versions in ubuntu 18.04 and 20.04 respectively). Secondly, for components like MIOpen and MIGraphX, we build with the cmake that comes with ubuntu 18.04, so we need to make sure it works with at least 3.10.

The question is if the 3.12 implementation is necessary as we could just use the implementation that works for 3.10? But maybe there is a functional difference.

Reply:
> Reordering the installation of NAMELINK_SKIP and NAMELINK_ONLY apparently fixed this issue.

So it seems it would be simpler to just use the 3.10 implementation instead of having two different versions here.

Reply:
I concur, just running a couple of further local tests to double check that the 3.10 implementation is functionally identical.